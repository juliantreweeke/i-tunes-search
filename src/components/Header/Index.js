import PropTypes from "prop-types";
import React from "react";
import Heading from '../Heading/Index';
import SearchBar from '../SearchBar/Index';
import styles from "./header.module.css";
import iTunesIcon from "../../itunes.svg";
import EN from "../../EN.json";

const Header = ({ children }) => {
  return (
    <header>
      <Heading headingLevel="h1">{EN.I_TUNES_ALBUM_SEARCH}</Heading>
      {children}
    </header>
  );
};

Header.propTypes = {
  children: PropTypes.node.isRequired,
};

export default Header;
