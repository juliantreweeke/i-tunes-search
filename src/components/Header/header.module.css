header {
  display: grid;
  align-items: center;
  width:100%;
}

header h1 {
  text-align: center;
  padding-top: var(--space-xs);
}

@media (min-width: 769px) {  
  header {
    grid-template-columns: 1fr 1fr;                         
  }

  header h1 {
    text-align: left;
  }
}
