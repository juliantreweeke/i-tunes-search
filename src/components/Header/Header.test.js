import React from "react";
import { render } from "@testing-library/react";
import Header from "./Index";
import { axe, toHaveNoViolations } from "jest-axe";

describe("Header", () => {
  it("should render", () => {
    const { container } = render(
      <Header>
        <div>
          <p>content</p>
        </div>
      </Header>
    );
    expect(container).toBeTruthy();
  });

  it("matches snapshot", () => {
    const { container } = render(
      <Header>
        <div>
          <p>content</p>
        </div>
      </Header>
    );
    expect(container).toMatchSnapshot();
  });

  it("component should have no accessibility violations", async () => {
    const { container } = render(
      <Header>
        <div>
          <p>content</p>
        </div>
      </Header>
    );
    expect.extend(toHaveNoViolations);
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });
});
