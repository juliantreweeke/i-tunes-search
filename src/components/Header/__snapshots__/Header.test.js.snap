// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <h1
      class="heading"
      data-size="xl"
    >
      iTunes Albums
    </h1>
    <div>
      <p>
        content
      </p>
    </div>
  </header>
</div>
`;
